### PR TITLE
[OKD-Core] Add Cluster Overview

### PR DESCRIPTION
**Pull request by rawagner**
Depends on: https://github.com/kubevirt/web-ui-components/pull/247

### CODE REVIEW

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    });
+  }
+
+  _setConsumersCpuData(response) {
+    console.log('updated cpu data');
+    const result = response.data.result;
+    this.setState(state => ({
+      consumersData: {
+        metrics: {
+          ...(_.get(state.consumersData, 'metrics') || {}),
```
Comment (mareklibra) [inline frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: The third arg of the `_.get()` is `defaultValue`

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+export class ClusterOverview extends React.Component {
+  constructor(props){
+    super(props);
+    this.messages = {};
+    this.state = {
+      healthData: {
+        data: {},
+        loaded: false,
+      },
+      consumersData: {
```
[the remark this answers is not in the document]
Comment (suomiy) [reply frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: another try: `topConsumerMetrics`?

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+
+  fetchHealth(callback) {
+    coFetch(`${k8sBasePath}/healthz`)
+      .then(response => response.text())
+      .then(text => text === 'ok' ? callback(true, 'All systems healthy') : callback(false, text))
+      .catch(() => callback(false, 'Cannot get cluster health'))
+      .then(() => {
+        const timeoutId = setTimeout(() => this.fetchHealth(callback), REFRESH_TIMEOUT);
+        this.intervals.push(timeoutId);
```
Comment (mareklibra) [inline frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: The array can become extremely huge

File: frontend/public/components/events.jsx [hunk 4]
```diff
@@ -467,14 +467,16 @@ class EventStream extends SafetyFirst {
 
     return <div className="co-m-pane__body co-m-pane__body--alt">
       <div className="co-sysevent-stream">
-        <div className="co-sysevent-stream__status">
-          <div className="co-sysevent-stream__timeline__btn-text">
-            { statusBtnTxt }
+        {!this.props.overview &&
```
Comment (mareklibra) [inline frontend/public/components/events.jsx]: Vivat merge conflicts!

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  utilizationStats,
+  capacityStats,
+};
+
+const REFRESH_TIMEOUT = 30000;
+
+const fetchPrometheusQuery = (query, callback) => {
+  const promURL = window.SERVER_FLAGS.prometheusBaseURL;
+  const url = `${promURL}/api/v1/query?query=${encodeURIComponent(query)}`;
+  coFetchJSON(url).then(callback).then(() => setTimeout(() => fetchPrometheusQuery(query, callback), REFRESH_TIMEOUT));
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: @suomiy fixed - setTimeout is not called when component is unmounted

File: frontend/public/components/events.jsx [hunk 6]
```diff
@@ -33,6 +33,7 @@ import {
   Timestamp,
   TogglePlay,
 } from './utils';
+import { EventsInnerOverview } from '../kubevirt/components/cluster/events-inner-overview';
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/public/components/events.jsx]: @suomiy as we discussed - Im passing Inner component as prop, defaults to original <Inner>

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+  if (resources.vms) {
+    inventory.vms = {
+      data: resources.vms,
+      title: 'VMs',
+      kind: VirtualMachineModel.kind,
+    };
+  }
+  props.inventoryData = {
+    inventory,
+    loaded: !!inventory,
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: nitpicking: `isLoaded`

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+  fetchPrometheusQuery(query, callback) {
+    const promURL = window.SERVER_FLAGS.prometheusBaseURL;
+    const url = `${promURL}/api/v1/query?query=${encodeURIComponent(query)}`;
+    coFetchJSON(url).then(result => {
+      if (this._isMounted) {
+        callback(result);
+      }
+    }).then(() => {
+      if (this._isMounted) {
```
Comment (mareklibra) [inline frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: I believe this is good enough

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: as we discussed, every component will have a different logic for resolving if all data are loaded or not and we will remove this loaded prop

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+        metrics: {
+          cpu: {
+            title: 'CPU',
+            consumers: [],
+          },
+        },
+        loaded: false,
+      },
+    };
+    this.setConsumersData = this._setConsumersData.bind(this);
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: tectonic rules do not allow using labdas in classes for some reason..

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+  constructor(props){
+    super(props);
+    this.messages = {};
+    this.state = {
+      healthData: {
+        data: {},
+        loaded: false,
+      },
+      consumersData: {
+        metrics: {
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: We will change these as we implement overview cards and find out what are the needs.

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (suomiy) [reply frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: ah, ok

File: frontend/public/kubevirt/components/cluster/cluster-overview.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (suomiy) [reply frontend/public/kubevirt/components/cluster/cluster-overview.jsx]: I would prefer to cancel the timeout immediatelly but this is good enough, I guess

**Comment by suomiy** [comment]
merging with a refactoring in mind which should come soon

**Pull request merged**